agent/configentry/merge_service_config.go

```diff
@@ -161,17 +161,31 @@ func MergeServiceConfig(defaults *structs.ServiceConfigResponse, service *struct
 	// remoteUpstreams contains synthetic Upstreams generated from central config (service-defaults.UpstreamConfigs).
 	remoteUpstreams := make(map[structs.PeeredServiceName]structs.Upstream)
 
+	// If the arguments did not fully normalize tenancy stuff, take care of that now.
+	entMeta := ns.EnterpriseMeta
+	entMeta.Normalize()
+
 	for _, us := range defaults.UpstreamConfigs {
 		parsed, err := structs.ParseUpstreamConfigNoDefaults(us.Config)
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse upstream config map for %s: %v", us.Upstream.String(), err)
 		}
 
-		remoteUpstreams[us.Upstream] = structs.Upstream{
-			DestinationNamespace: us.Upstream.ServiceName.NamespaceOrDefault(),
-			DestinationPartition: us.Upstream.ServiceName.PartitionOrDefault(),
-			DestinationName:      us.Upstream.ServiceName.Name,
-			DestinationPeer:      us.Upstream.Peer,
+		// If the defaults did not fully normalize tenancy stuff, take care of
+		// that now too.
+		psn := us.Upstream // only normalize the copy
+		psn.ServiceName.EnterpriseMeta.Normalize()
+
+		// Normalize the partition field specially.
+		if psn.Peer != "" {
```

Comment:
A peered upstream MUST exist in the same partition as the downstream using it. When it is empty, inferring it as `default` is incorrect.